blog/2024-11-mastering-effect-and-afterRenderEffect/README.md

```diff
@@ -0,0 +1,477 @@
+The flag was planned as an exception, but it was too often used in legitimate cases where setting signals was reasonable or even necessary, such as updating a signal after a series of changes or working with multiple signals.
+In response, Angular’s new approach now allows setting signals within `effect()` by default, removing the need for `allowSignalWrites`.
+This more flexible design reflects Angular’s commitment to simplifying the development experience.
+See the [official blog post](https://blog.angular.dev/latest-updates-to-effect-in-angular-f2d2648defcd) that confirms this new guidance.
+
+We interpret this new information in the following way:
+> 💡 **It is now a valid case to use `effect()` for state updates or side effects that are difficult to achieve with other reactive primitives, such as `computed()`**.
+
+This change to the paradigm is in line with new features introduced in Angular 19, such as `linkedSignal()` and `resource()`.
+Both help to maintain cleaner and more declarative state management patterns where possible. 
+Good patterns are no longer enforced by the `allowSignalWrites` flag, but instead by useful high-level signal APIs.
+
+With this shift, here’s a new general rule of thumb:
+- **Use `effect()`** for tasks traditionally performed in `ngOnInit` or `ngOnChanges`.
+- **Use `afterRenderEffect()`** for tasks traditionally handled in `ngAfterViewInit` or `ngAfterViewChecked`, or when you need to interact directly with rendered DOM elements.
+
+Let’s dive into the specifics! 🚀
+
+
+
+## Core Differences Between `effect()` and `afterRenderEffect()`
+
+Both `effect()` and `afterRenderEffect()` are designed to track and respond to changes in signals, but they differ in timing and use cases.
+
+- **`effect()`** runs as part of the Angular change detection cycle and can now safely modify signals without any additional flags.
+- **`afterRenderEffect()`** is a lower-level API that executes after the DOM has been updated. 
+  It’s particularly suited for tasks that require interacting directly with the DOM, such as measuring element sizes or making complex visual updates in phases.
+
+Here’s a simple comparison to illustrate how these functions operate:
+
+```typescript
+counter = signal(0);
+
+effect(() => {
+  console.log(`Current counter value: ${this.counter()}`);
+});
+
+afterRenderEffect(() => {
+  console.log('DOM rendering completed for this component');
+});
+```
+
+As expected, the console output for `afterRenderEffect` is triggered after the output of `effect`.
+
+
+## Introducing `effect()`
+
+Angular supports two types of effects: **component effects** and **root effects**. 
+Component effects are initiated when `effect()` is called within a component, directive, or a service tied to them. 
+Root effects, on the other hand, are initiated when `effect()` is called outside the component tree, such as in a root service, or by setting the `forceRoot` option.
```

Comment:
What is a root service? 

```suggestion
Root effects, on the other hand, are initiated when `effect()` is called outside the component tree, such as in a root service, or by setting the `forceRoot` option.
```

Reply:
Well yeah, but what is it anyway? Even a service provided in the root injector will only be instanced when a component injects it.